### PR TITLE
added findMaxRaise

### PR DESCRIPTION
**Pull request by HelgeWieding**
Function to find max raise

### CODE REVIEW

File: lib/helper.spec.js [hunk 1]
```diff
@@ -1276,5 +1276,69 @@ describe('getBbPos() ', () => {
...
+        last: new EWT(ABI_BET).bet(1, 200).sign(P1_KEY)
+      }, {
+        address: P2_ADDR,
+        last: new EWT(ABI_BET).bet(1, 500).sign(P2_KEY)
+      }, {
+        address: P3_ADDR,
+        last: new EWT(ABI_BET).bet(1, 1200).sign(P3_KEY)
+        }, {
+        address: P4_ADDR,
+        last: new EWT(ABI_BET).bet(1, 100).sign(P3_KEY)
```
Comment (johannbarbie) [inline lib/helper.spec.js]: should be signed by P4

File: lib/helper.js [hunk 2]
```diff
@@ -135,6 +135,23 @@ PokerHelper.prototype.nextActivePlayer = function(lineup, playerPos) {
   return newPos;
 }
 
+PokerHelper.prototype.findMaxRaise = function(lineup, dealer, lastRoundMaxBet) {
```
Comment (johannbarbie) [inline lib/helper.js]: do input validation

File: lib/helper.js [hunk 3]
```diff
@@ -135,6 +135,23 @@ PokerHelper.prototype.nextActivePlayer = function(lineup, playerPos) {
...
+PokerHelper.prototype.findMaxRaise = function(lineup, dealer, lastRoundMaxBet) { 
+  const max = this.findMaxBet(lineup, dealer);
+  if (max.amount === lastRoundMaxBet) {
+    return 0;
+  } 
+  // finding second highest bet
+  let secondMax = lastRoundMaxBet;
+  for (let i = max.pos + 1; i < lineup.length + max.pos; i++) {
+    const pos = i % lineup.length;
+    const last = (lineup[pos].last) ? this.rc.get(lineup[pos].last).values[1] : 0;
```
Comment (johannbarbie) [inline lib/helper.js]: put some empty seats into tests to cover this line.

File: lib/helper.js (hunk 2 above)
Comment (johannbarbie) [inline lib/helper.js]: better name would be "findMinRaiseAmount"

Comment (johannbarbie) [inline lib/helper.js]: min raise amount if no-one raised before? SB? BB?

File: lib/helper.js [hunk 4]
```diff
@@ -135,6 +135,23 @@ PokerHelper.prototype.nextActivePlayer = function(lineup, playerPos) {
   return newPos;
 }
 
+PokerHelper.prototype.findMaxRaise = function(lineup, dealer, lastRoundMaxBet) { 
+  const max = this.findMaxBet(lineup, dealer);
+  if (max.amount === lastRoundMaxBet) {
+    return 0;
```
Comment (johannbarbie) [inline lib/helper.js]: return value 0 doesn't make sense, raise with 0 should not be allowed.

File: lib/helper.js (hunk 2 above)
Comment (HelgeWieding) [inline lib/helper.js]: then it would a bet and it would be the big blind. Or the your stacksize if you dont have the big blind.

Comment (johannbarbie) [inline lib/helper.js]: i would either pass in the BB as param, and return it, or return -1, saying that it is not this functions scope to determine this case.

Comment (HelgeWieding) [inline lib/helper.js]: fixed

File: lib/helper.js (hunk 4 above)
Comment (HelgeWieding) [inline lib/helper.js]: fixed

File: lib/helper.js (hunk 3 above)
Comment (HelgeWieding) [inline lib/helper.js]: done

File: lib/helper.spec.js (hunk 1 above)
Comment (HelgeWieding) [inline lib/helper.spec.js]: fixed

**Pull request merged**